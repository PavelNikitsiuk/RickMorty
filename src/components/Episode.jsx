import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import styles from './Episodes.module.css';

class Episode extends PureComponent {
	static propTypes = {
		episodeData: PropTypes.shape({
			name: PropTypes.string,
			episode: PropTypes.string
		})
	};

	render() {
		const {episodeData} = this.props;

		return (
			<div className={styles.button}>
				<div className={styles.name}>{episodeData.name}</div>
				<div>{episodeData.episode}</div>
			</div>
		);
	}
}

export default Episode;
