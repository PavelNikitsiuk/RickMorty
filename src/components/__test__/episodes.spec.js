import React from 'react'
import Episode from "../Episode";
import { shallow } from 'enzyme'

describe('Episode', () => {
	const props = {
		episodeData: {
			name: "string",
			episode: "string"
		}
	};

	const shallowEpisode = shallow(<Episode {...props} />);

	it('Episode renders properly', () => {
		expect(shallowEpisode).toMatchSnapshot();
	});
});