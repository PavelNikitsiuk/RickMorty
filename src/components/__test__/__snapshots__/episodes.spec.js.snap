// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode Episode renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Episode
    episodeData={
      Object {
        "episode": "string",
        "name": "string",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="name"
        >
          string
        </div>,
        <div>
          string
        </div>,
      ],
      "className": "button",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "string",
          "className": "name",
        },
        "ref": null,
        "rendered": "string",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "string",
        },
        "ref": null,
        "rendered": "string",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="name"
          >
            string
          </div>,
          <div>
            string
          </div>,
        ],
        "className": "button",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "string",
            "className": "name",
          },
          "ref": null,
          "rendered": "string",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "string",
          },
          "ref": null,
          "rendered": "string",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
